### PR TITLE
Add a lot more leak checks and fix some leaks

### PR DESCRIPTION
**Pull request by howardjohn**
First, move all leak check main to dedicate files. This makes it much easier to see which ones have leak check enabled, and also avoids accidentally deleting the check when its in a random spot in some file. Can also search for all packages without leak checker enable with: `find pilot -type d '!' -exec test -e "{}/leak_test.go" \; -exec bash -c 'echo  "$(ls {} | grep test.go | wc -l) {}"' \; | sort -n`

Next, add the leak check to a lot of important packages and fix a lot of leaks. All of these are about just not stopping properly/

### CODE REVIEW

File: pilot/pkg/bootstrap/server_test.go [hunk 1]
```diff
@@ -380,7 +380,7 @@ func TestIstiodCipherSuites(t *testing.T) {
 		},
 	}
 
-	for _, c := range cases {
+	for _, c := range cases[:1] {
```
Comment (stevenctl) [inline pilot/pkg/bootstrap/server_test.go:383]: not intended to merge?

Comment (howardjohn) [reply pilot/pkg/bootstrap/server_test.go:383]: thanks!

File: pilot/pkg/serviceregistry/kube/controller/controller_test.go [hunk 2]
```diff
@@ -162,6 +162,7 @@ func TestServices(t *testing.T) {
 				t.Fatalf("GetService(%q) => %s, should not exist", missing, svc.Hostname)
 			}
 		})
+		return
```
Comment (stevenctl) [inline pilot/pkg/serviceregistry/kube/controller/controller_test.go:165]: same here?

**Review by douglas-reid — APPROVED** [review]
The changes to the wg-policies-and-telemetry-maintainers owned files LGTM.

**Review by lei-tang — APPROVED** [review]

**Review by stevenctl — APPROVED** [review]

**Pull request merged**